src/utils/market.util.ts

```diff
@@ -0,0 +1,140 @@
+import { MarketCoinsSortFieldEnum } from '../enums/market-coins-sort-field.enum';
+import { MarketCoin, MarketCoinRaw } from '../store/market/market.interfaces';
+import { Colors } from '../styles/colors';
+import { coingeckoApi, templeWalletApi } from './../api.service';
+
+export const fetchMarketTopCoins = () =>
+  coingeckoApi
+    .get<Array<MarketCoinRaw>>(
+      'coins/markets?vs_currency=usd&category=tezos-ecosystem&order=market_cap_desc&per_page=100&page=1&sparkline=false&price_change_percentage=24h%2C7d'
+    )
+    .then(({ data }) => {
+      const mappedCoins: Array<MarketCoin> = [];
+
+      for (const coinInfo of data) {
+        mappedCoins.push({
+          id: coinInfo.id,
+          name: coinInfo.name,
+          symbol: coinInfo.symbol.toUpperCase(),
+          imageUrl: coinInfo.image,
+          price: coinInfo.current_price,
+          priceChange7d: coinInfo?.price_change_percentage_7d_in_currency,
+          priceChange24h: coinInfo?.price_change_percentage_24h_in_currency,
+          volume24h: coinInfo.total_volume,
+          supply: coinInfo.circulating_supply,
+          marketCup: coinInfo.market_cap
+        });
+      }
+
+      return mappedCoins;
+    });
+
+export const fetchMarketCoinsSlugs = () =>
+  templeWalletApi.get<Record<string, string>>('/top-coins').then(value => value.data);
+
+const si = [
+  { value: 1e3, letter: 'K' },
+  { value: 1e6, letter: 'M' },
+  { value: 1e9, letter: 'B' },
+  { value: 1e12, letter: 'T' },
+  { value: 1e15, letter: 'P' },
+  { value: 1e18, letter: 'E' }
+];
+
+const formatNumber = (value: number) => {
+  if (value < 10000) {
+    if (value % 1 === 0) {
+      return value.toFixed(0);
+    }
+
+    return value.toFixed(2);
+  }
+
+  let index;
+  for (index = si.length - 1; index > 0; index--) {
+    if (value >= si[index].value) {
+      break;
+    }
+  }
+
+  return (value / si[index].value).toFixed(2).replace(/\.0+$|(\.[0-9]*[1-9])0+$/, '$1') + si[index].letter;
+};
```

Comment:
maybe extend our `kFormatter` function with 'M' and 'B', and this will be enough for us?  
cuz this function looks big and scarry